backends/p4fmt/p4fmt.cpp

```diff
@@ -23,6 +24,9 @@ std::stringstream getFormattedOutput(std::filesystem::path inputFile) {
     }
 
     auto top4 = P4::ToP4(&formattedOutput, false);
+    auto attach = P4::Attach(&formattedOutput, false);
+    // attach comments to nodes
+    program->apply(attach);
```

Comment:
Note that you can do this in one line as

    program->apply(P4::Attach(&formattedOutput, false));  // attached comments to nodes

When you have more than one pass to apply, it may make sense to combine them into a PassManager:

    PassManager passes({
        P4::Attach(&formattedOutput, false),
        P4::ToP4(&formattedOutput, false)
    });
    program = program->apply(&passes);

assigning the result of `apply` back to `program` is unneeded IF all the passes are just Inspectors, but doesn't hurt.

Reply:
@snapdgn whas this addressed? I do not see anything here.

Reply:
This was addressed, was was changed back to not using pass manager based on another suggestion: https://github.com/p4lang/p4c/pull/4845#discussion_r1726480839